app/code/community/Bolt/Boltpay/Controller/Traits/ApiControllerTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+/**
+ * Trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait
+ *
+ * Defines generalized actions associated with API calls to and from the Bolt server
+ *
+ * @method Mage_Core_Controller_Response_Http getResponse()
+ * @method Mage_Core_Model_Layout getLayout()
+ */
+trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait {
+    use Bolt_Boltpay_BoltGlobalTrait;
+    /**
+     * @var string The body of the request made to this controller
+     */
+    protected $payload;
+    /**
+     * @var string The signed payload which used the stores signing secret
+     */
+    protected $signature;
+    /**
+     * @var bool determines if JSON is expected return type for preDispatch optimization.
+     */
+    protected $willReturnJson = true;
+    /**
+     * @var bool mandates that all request to this controller must be signed
+     */
+    protected $requestMustBeSigned = true;
```

Comment:
same as above; this must always be true

Reply:
If this was always true, I wouldn't include it.  There are various instances requiring different values, especially when thinking long-term
- Must be signed - Web Hook, Shipping and Tax,  Onepage Controller (OPC) Success Page
- Must be accessible without being signed - OPC all other endpoints, OPC non-Bolt and 1.x Bolt Success Page

Reply:
Actually, I agree with this statement.  You are certainly right that something exhibiting webhook will always need to be signed.  I will update the code and the trait to be as such.  Also, in thinking about how people often override OnePage Checkout, I am going to restructure the override that happens to utilize traits in a way that makes it very simple for integration to extend without overriding any core methods.